pkg/controller/service/servicecontroller.go

```diff
@@ -488,6 +491,9 @@ func (s *ServiceController) needsUpdate(oldService *api.Service, newService *api
 			return true
 		}
 	}
+	if !reflect.DeepEqual(oldService.Annotations, newService.Annotations) {
```

Comment:
They are maps. Order does not matter here. 
http://stackoverflow.com/questions/18208394/testing-equivalence-of-maps-golang